### ISSUE TITLE
Sinatra::Namespace breaks route with named parameters ?

### ISSUE THREAD

**Issue by scalp42** [sinatra/sinatra#1409]
Hi folks,

Running into an issue upgrading to Sinatra 2.

This route works fine without `Sinatra::Namespace`:

```ruby
class TestController < MainController
  get '/i/love/:framework.?:format?' do
    params.to_json
  end
end
```

```
$> http --body localhost:9393/v1/i/love/sinatra

{
    "captures": [],
    "format": null,
    "framework": "sinatra"
}

$> http --body localhost:9393/v1/i/love/sinatra.json

{
    "captures": [],
    "format": "json",
    "framework": "sinatra"
}
```

If I try to namespace the same route with `Sinatra::Namespace`:

```ruby
class TestController < MainController
  register Sinatra::Namespace

  namespace '/hey' do
    get '/i/love/:framework.?:format?' do
      params.to_json
    end
  end
end
```

It won't start:

```
$> bundle exec puma --port=9393 config.ru

Puma starting in single mode...
* Version 3.11.3 (ruby 2.5.0-p0), codename: Love Song
* Min threads: 2, max threads: 10
* Environment: development

! Unable to load application: Mustermann::Error: #<Class:0x00007fb6e10c22a0>: Cannot translate Mustermann::AST::Node::WithLookAhead
```

Here is the full trace: https://gist.github.com/scalp42/e627e470c6169e3f4be380cea327b697

Any ideas? 🤔

Thanks in advance for the help!

**Comment by namusyaka** [comment]
It's almost certainly a bug. I'll investigate this problem soon.
Thank you for the report.

**Comment by scalp42** [comment]
No worries, thanks for looking into it! 🥇

**Comment by namusyaka** [comment]
This is not sinatra issue, but mustermann issue.
The cause of this issue is to not process `WithLookAhead` node in `Mustermann::Sinatra::SafeRenderer`.
I sent [a PR](https://github.com/sinatra/mustermann/pull/86) for fixing the issue.

@scalp42 Could you confirm it? Maybe the following code can be used.

```ruby
gem 'mustermann', github: 'sinatra/mustermann', branch: 'fix-sinatra-concat'
```

**Comment by namusyaka** [comment]
@scalp42 Thanks! This bug could be fixed at the next mustermann release. https://github.com/sinatra/mustermann/issues/87

**Comment by scalp42** [comment]
@namusyaka @rkh any idea when a new mustermann version will be released with the fix?

Thanks in advance!

**Comment by namusyaka** [comment]
@scalp42 Sorry for the late reply, I have just released mustermann [v1.0.3](https://rubygems.org/gems/mustermann/versions/1.0.3). Please check it out.
